src/poetry/puzzle/provider.py

```diff
@@ -635,23 +628,18 @@ def complete_package(self, package: DependencyPackage) -> DependencyPackage:
                 markers.append(deps[0].marker)
 
             _deps = [_dep[0] for _dep in by_constraint.values()]
-            self.debug(
-                "<warning>Different requirements found for {}.</warning>".format(
-                    ", ".join(
-                        "<c1>{}</c1> <fg=default>(<c2>{}</c2>)</> with markers <b>{}</b>".format(
-                            d.name,
-                            d.pretty_constraint,
-                            d.marker if not d.marker.is_any() else "*",
-                        )
-                        for d in _deps[:-1]
-                    )
-                    + " and "
-                    + "<c1>{}</c1> <fg=default>(<c2>{}</c2>)</> with markers <b>{}</b>".format(
-                        _deps[-1].name,
-                        _deps[-1].pretty_constraint,
-                        _deps[-1].marker if not _deps[-1].marker.is_any() else "*",
-                    )
+            warnings = (
+                ", ".join(
+                    f"<c1>{d.name}</c1> <fg=default>(<c2>{d.pretty_constraint}</c2>)</> "
+                    f"with markers <b>{d.marker if not d.marker.is_any() else '*'}</b>"
```

Comment:
I'd like to get this conditional out of the string and DRY it up. I'm thinking a lambda saved in a variable might be the right choice -- thoughts?

Reply:
factored out the conditional, and then realized the entire string could be factored out. went with a standard function over lambda due to https://www.flake8rules.com/rules/E731.html